controllers/dynamic_resource_request_controller.go

```diff
@@ -311,59 +242,59 @@ func (r *dynamicResourceRequestController) deleteWork(ctx context.Context, resou
 		if errors.IsNotFound(err) {
 			// only remove finalizer at this point because deletion success is guaranteed
 			controllerutil.RemoveFinalizer(resourceRequest, finalizer)
-			if err := r.Client.Update(ctx, resourceRequest); err != nil {
+			if err := r.Client.Update(o.ctx, resourceRequest); err != nil {
 				return err
 			}
 			return nil
 		}
 
-		logger.Error(err, "Error locating Work, will try again in 5 seconds", "workName", workName)
+		o.logger.Error(err, "Error locating Work, will try again in 5 seconds", "workName", workName)
 		return err
 	}
 
-	err = r.Client.Delete(ctx, work)
+	err = r.Client.Delete(o.ctx, work)
 	if err != nil {
 		if errors.IsNotFound(err) {
 			// only remove finalizer at this point because deletion success is guaranteed
 			controllerutil.RemoveFinalizer(resourceRequest, finalizer)
-			if err := r.Client.Update(ctx, resourceRequest); err != nil {
+			if err := r.Client.Update(o.ctx, resourceRequest); err != nil {
 				return err
 			}
 			return nil
 		}
 
-		logger.Error(err, "Error deleting Work %s, will try again in 5 seconds", "workName", workName)
+		o.logger.Error(err, "Error deleting Work %s, will try again in 5 seconds", "workName", workName)
 		return err
 	}
 
 	return nil
 }
 
-func (r *dynamicResourceRequestController) deleteWorkflows(ctx context.Context, resourceRequest *unstructured.Unstructured, resourceRequestIdentifier, finalizer string, logger logr.Logger) error {
+func (r *dynamicResourceRequestController) deleteWorkflows(o opts, resourceRequest *unstructured.Unstructured, resourceRequestIdentifier, finalizer string) error {
 	jobGVK := schema.GroupVersionKind{
 		Group:   batchv1.SchemeGroupVersion.Group,
 		Version: batchv1.SchemeGroupVersion.Version,
 		Kind:    "Job",
 	}
 
-	jobLabels := pipeline.Labels(resourceRequestIdentifier, r.promiseIdentifier)
+	jobLabels := pipeline.LabelsForAllResourceWorkflows(resourceRequestIdentifier, r.promiseIdentifier)
 
-	resourcesRemaining, err := deleteAllResourcesWithKindMatchingLabel(ctx, r.Client, jobGVK, jobLabels, logger)
+	resourcesRemaining, err := deleteAllResourcesWithKindMatchingLabel(o, jobGVK, jobLabels)
 	if err != nil {
 		return err
 	}
 
 	if !resourcesRemaining {
 		controllerutil.RemoveFinalizer(resourceRequest, finalizer)
-		if err := r.Client.Update(ctx, resourceRequest); err != nil {
+		if err := r.Client.Update(o.ctx, resourceRequest); err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
 
-func (r *dynamicResourceRequestController) setStatus(rr *unstructured.Unstructured, logger logr.Logger, statuses ...string) {
+func setStatus(rr *unstructured.Unstructured, logger logr.Logger, statuses ...string) {
```

Comment:
agree that this doesn't need to be part of the struct; should this (and some of the other functions that got removed from the struct) move to one of the shared libs 🤔? those package-global functions can get hard to follow/find sometimes

Reply:
thats a good shout. I'll see what can be moved out

Reply:
moved a bunch, lmk what you think